spec/requests/api/my_request/requests/create_spec.rb

```diff
@@ -133,18 +133,59 @@
                title: 'reQuest title',
                description: 'You shall come and help me!',
                reward: 100,
-               category: 'car'
+               category: 'car',
+               coords: { long: 55.3, lat: 32.1 }
              }
         @quest = Request.last
       end
-      
+
       it 'has 422 response' do
         expect(response).to have_http_status 422
       end
-  
+
       it 'responds with an error message' do
         expect(response_json['message']).to eq "'car' is not a valid category"
       end
     end
+
+    describe 'no location is provided' do
+      before do
+        post '/api/my_request/requests',
+             headers: headers,
+             params: {
+               title: 'reQuestus title',
+               description: 'You shall come and help me!',
+               reward: 200
+             }
+      end
+      it 'has 422 response' do
+        expect(response).to have_http_status 422
+      end
+
+      it 'responds with an error message' do
+        expect(response_json['message']).to eq "Location can't be blank"
```

Comment:
If visitor does not share location, the coords will be sent as coords { lat: undefined, long: undefined} from front end.
For future reference, if visitor not sharing location the Back end should serve the request list  distance:null to each one of them. So the distance will not be shown to that user. 
That's how we did it in the front end